README.md

```diff
@@ -13,7 +13,15 @@ Install the package:
 npm i webpack-serve-overlay
 ```
 
-Then require the overlay at the top of your `index.jsx` (or equivalent):
+Then add the following entry to the [`entry` object](https://webpack.js.org/configuration/entry-context/#entry) in your webpack config:
```

Comment:
I've just tested doing this, and it seems to stop hot-reloading: 

```
    entry: {
        overlay: 'webpack-serve-overlay',
        bundle: path.resolve('src/index.jsx')
    },
```

Results in the following console.logs:

```
｢hot｣ webpack: Compiling (<unnamed compiler>)
10:15:42.412 log.js?60b4:59 ｢hot｣ App Updated, Reloading Modules
10:15:42.412 log.js?60b4:59 ｢hot｣ Checking for updates to the bundle.
10:15:42.488 log.js?60b4:59 ｢hot｣ No Modules Updated.
10:15:42.489 log.js?60b4:59 ｢hot｣ App is up to date.
```

If I remove the entry, and then make the same change, the page gets reloaded. Could you check this out, and see if you can reproduce this on your end?

Reply:
So I definitely can't reproduce that error with my config and [my boilerplate](https://github.com/agilgur5/front-end-base/) is now set up [similarly](https://github.com/agilgur5/front-end-base/blob/master/webpack.config.js) (slightly different from the [branch](https://github.com/agilgur5/front-end-base/blob/only-if-production/webpack.config.js) linked above). You can try running that and changing some code to see if it's potentially related to your environment.

Logs ok
```
log.js:59 ｢hot｣ Hot Module Replacement Enabled. Waiting for signal.
log.js:59 ｢hot｣ WebSocket connected
log.js:59 ｢hot｣ App updated. Recompiling components/app.js
log.js:59 ｢hot｣ webpack: Compiling (<unnamed compiler>)
log.js:59 ｢hot｣ App Updated, Reloading Modules
log.js:59 ｢hot｣ Checking for updates to the bundle.
log.js:59 ｢hot｣ No Modules Updated.
log.js:59 ｢hot｣ App is up to date.
````

Kinda hard to tell without more context (e.g. the rest of the config) what might be causing that issue for you. Wasn't totally clear from your comment, but just to make sure webpack can't currently detect changes in its own config

Reply:
That's what I expected; I'll provide more context soon, when I have some free time on my hands.

And don't worry, I know that you have to restart webpack for changes to take effect ;) 

Heres a gist containing how my webpack configs are setup:

https://gist.github.com/G-Rath/b441cd818940ff989edb216513b8d4f3

For the record, all of my configuration stuff was/is based off the examples given in the Webpack documentation, which is somewhat concerning if it's not compatible.

Reply:
Huh ok. There are like a 100 different ways of doing the same thing in webpack and before 2.0 the docs weren't the best place to find info but that's surprising nowadays ¯\_(ツ)_/¯ . 
Your config is a bit more complex than I expected, so not sure staring for errors would work well with it haha. If you have a `package.json` and a sample js file that replicates the problem (like the one in my boilerplate) I can test and see if it's reproduced in my environment, otherwise in your case I'd try to remove a bunch of the config and see if it still occurs.
Were you saying that HMR works with `path.resolve` but not the plain package name?? Because the gist isn't using `onlyIfDevelopment` so that's unrelated

(Indeed, another fun fact is that syntax causes problems on certain platforms like React Native because they have some "optimizations" that require you to have "a value" inserted there... I didn't investigate further into it but it was quite disorienting so I stuck to the more explicit syntax instead and kept using it everywhere)

Reply:
@G-Rath so I started getting the problem of not hot-reloading as well, and found out that it's because ["`webpack-hot-client` automatically configur[es] the client script for entries on the _first_ entry only"](https://github.com/webpack-contrib/webpack-serve/issues/119#issuecomment-401502247).

I do think it's a significant enough and common enough problem (particularly based on all the confusion in that issue) that it's worth adding a note to the docs about, so I'll amend my PR with that.